config.json

```diff
@@ -2379,6 +2379,25 @@
           "pattern-matching"
         ],
         "difficulty": 9
+      },
+      {
+        "slug": "zebra-puzzle",
+        "name": "Zebra Puzzle",
+        "uuid": "0a561b08-6e61-4415-a3f1-e7d411a4a0e2",
+        "prerequisites": [
+          "pattern-matching",
+          "case",
+          "cond",
+          "if",
+          "enum",
+          "keyword-lists",
```

Comment:
I didn't see keyword lists in your solution, but I did see maps. Do you think we should swap this prerequisite for maps? Or should we have both? I'm also seeing recursion in your solution. Maybe that should also be a prerequisite?

A prerequisite doesn't mean you have to use that concept to solve the exercise, but it will stop people from solving it before they do know the concept (so that they can properly choose, for example enum vs recursion, or both).

@neenjaw thoughts?

Reply:
You're right. I did sort of use keyword lists at some point because when you enumerate on a map it turns into a keyword list, but I changed the implementation to use `Map.new()` instead, so I should remove this. I will swap the requirement with `map`.

Reply:
Yes, I think swapping the pre-req with maps is a good step for this.  While you are right, enumerated maps and key-word lists elements appear similarly I would hesitate to say that maps are transformed into keyword lists, the tuple-pair is the focus of that I think.

Personally I am okay with hard exercises appearing early in the exercise progression as long as the student has the tools to solve it.  So i would just make sure that there isn't a hidden pattern in the code required to solve that would be taught more explicitly with another more "advanced" concept